### DISCUSSION TITLE
Getting a 500 error with this Docker run command

### DISCUSSION THREAD

**Discussion by DeflateAwning** [open-webui/open-webui#2417] [General]
# Bug Report

## Description

**Bug Summary:**
```bash
# this one works fine, but I can't connect to Ollama, so I'm trying the next one (per the README troubleshooting):
docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main

# this results in an error:
docker run -d --network=host -v open-webui:/app/backend/data -e OLLAMA_BASE_URL=http://127.0.0.1:11434 --name open-webui --restart always ghcr.io/open-webui/open-webui:main
```

**Steps to Reproduce:**
1. Run docker commands above.

**Expected Behavior:**
I expect the page to load.

**Actual Behavior:**
I get a 500 error with no description of the error. The `docker log` output doesn't show anything useful either (no logs related to the error or attempted page load).

## Environment

- **Open WebUI Version:** 0.1.125
- **Ollama (if applicable):** 0.1.38

- **Operating System:** Ubuntu
- **Browser (if applicable):** N/A

## Installation Method

See above.

**Comment by haricalzi** [comment]
I have faced the same problem, read [here](https://github.com/open-webui/open-webui/discussions/2315#discussion-6677184) for a further explaination.

**Comment by thhung** [comment]
How do we solve this problem? I am still facing the same issue.

**Reply by randoentity** [reply]
I have a similar problem, but mine is likely just a bad reverse proxy configuration. But perhaps they stem from the same source.
I am using the github install method (no docker) with nginx (configured like https://github.com/open-webui/open-webui/discussions/1235#discussioncomment-9341387).

The workaround to go to /admin as linked above works for me, but refreshing results in the 500 error.

I might be able to debug further when I'm on a desktop browser. I'm not using Firefox nor Ubuntu like in the thread linked above though.

Also nothing but 200s in the backend logs.

Edit: it seems to be fixed by adding
```
proxy_set_header Host $host;
proxy_set_header X-Real-IP $remote_addr;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header X-Forwarded-Proto $scheme;
```
but it could just as well be a caching issue.

Edit: removed the directives above as I was still sometimes getting 500s, and without them it's the same. Sometimes it works and sometimes it doesn't. So now I'm thinking timeout issue, network issue, or race condition. In any case unlikely to be related to OP's issue.

**Reply by ghost** [reply]
Necroing this - my guess is it happens between container starts.  I have narrowed it down to a reproducible case.

1.  Start podman or docker image.  Poke around.

2.  podman kill CONTAINER_ID

3.  Run the same start script or run it in terminal

4.  500 error.  The problem is something isn't being cleared out properly on restart in your browser, so for now I'm just manually clearing cache.  I think you can do something to prevent the cache from persisting though, or at the very least rm -r CACHE FOLDER in a .sh script before you launch the container again.
